src/models/data/data-set.ts

```diff
@@ -620,15 +636,36 @@ export const DataSet = types.model("DataSet", {
         });
       },
 
-      selectAll(select = true) {
+      selectAllAttributes(select = true) {
+        if (select) {
+          clearAllSelections();
+          self.attributes.forEach(attribute => self.attributeSelection.add(attribute.id));
+        } else {
+          self.attributeSelection.clear();
+        }
+      },
+
+      selectAllCases(select = true) {
         if (select) {
-          self.cases.forEach(({__id__}) => self.selection.add(__id__));
+          clearAllSelections();
```

Comment:
Same as with `selectAllAttributes()` above, you shouldn't need to clear the case selection before selecting them all.